### PR TITLE
feat: add APIGateway support

### PR DESCRIPTION
**Pull request by glimberea**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: docs/data-sources/apigateway_route.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+subcategory: "ApiGateway"
+layout: "ionoscloud"
+page_title: "IonosCloud: route"
```
Comment (avirtopeanu-ionos) [inline docs/data-sources/apigateway_route.md:4]: I dont know if it matters, but this might need to be `IonosCloud: ionoscloud_apigateway_route`

Comment (glimberea) [reply docs/data-sources/apigateway_route.md:4]: changed it in https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/601

**Review by avirtopeanu-ionos — APPROVED** [review]
the check for logging /central looks great :+1:

File: go.mod [hunk 2]
```diff
@@ -13,7 +13,8 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.33.0
 	github.com/hashicorp/terraform-plugin-testing v1.8.0
 	github.com/iancoleman/strcase v0.3.0
-	github.com/ionos-cloud/sdk-go-bundle/products/logging/v2 v2.0.0
+	github.com/ionos-cloud/sdk-go-apigateway v0.0.0-20240618102110-09613e8cad5d
```
Comment (avirtopeanu-ionos) [inline go.mod:16]: Can you please do a `go get -u` to update to 1.0.0 for sdk-go-apigateway?

**Review by avirtopeanu-ionos — CHANGES_REQUESTED** [review]

**Review by cristiGuranIonos — DISMISSED** [review]

**Review by avirtopeanu-ionos — APPROVED** [review]

**Review by cristiGuranIonos — DISMISSED** [review]

**Review by avirtopeanu-ionos — APPROVED** [review]

**Pull request merged**